pyquil/api/_qam.py

```diff
@@ -60,18 +61,24 @@ def load(self, executable):
         return self
 
     @_record_call
-    def write_memory(self, *, region_name: str, offset: int = 0, value=None):
+    def write_memory(self, *, region_name: str, offset: int = 0, value: abc.Sequence = None):
```

Comment:
Actually if you're going to add a type hint, I'd recommend `from typing import Sequence`, then specify what the valid element types are here as well. Something like `Sequence[Union[int, float]]`. `typing.Sequence` inherits from `abc.Sequence`, so you can then nix the `abc` import and replace `abc.Sequence` with `Sequence` in the `isinstance` check, below.

Sorry for the flip-flopping. When I suggested `collections.abc.Sequence` in the other comment, I assumed you wouldn't be adding a type hint here. I *almost* added a parenthetical clause to my comment saying "prefer `typing.Sequence` if you want to add a type hint", but didn't want to muddle the point. Seems that backfired on me!

Reply:
Oh, actually I probably got that type hint wrong. I think `Union[Sequence[int], Sequence[float]]` is more correct, since a single memory region can only contain one of those types.